src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.cs

```diff
@@ -58,10 +192,75 @@ private static string ReadEnvironmentVariable(string key)
             }
             catch (Exception ex)
             {
-                StartupLogger.Log(ex, "Error while loading environment variable " + key);
+                StartupLogger.Log(ex, $"Error while loading environment variable \"{key}\".");
             }
 
             return null;
         }
+
+        private void ResolveManagedProfilerDirectory()
+        {
+            Assembly objectAssembly = (new object()).GetType().Assembly;
```

Comment:
why not `typeof(object).Assembly`?

Reply:
Good point. Updated in _macrogreg/loader-with-modules-cctor-and-managed-loader-branch_.